### PR TITLE
IPC Crayon Fix

### PR DESCRIPTION
**Pull request by Doxxmedearly**
IPCs can no longer enjoy the taste of crayons.

### CODE REVIEW

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -105,7 +105,7 @@
 	return
 
 /obj/item/pen/crayon/attack(mob/M as mob, mob/user as mob, var/target_zone)
-	if(M == user)
+	if(M == user && !isipc(user))
```
Comment (Alberyk) [inline code/game/objects/items/crayons.dm]: check_has_mouth is probably a better solution here than checking for species.

Comment (Doxxmedearly) [reply code/game/objects/items/crayons.dm]: hopefully did that right

File: code/game/objects/items/crayons.dm [hunk 2]
```diff
@@ -105,7 +105,7 @@
 	return
 
 /obj/item/pen/crayon/attack(mob/M as mob, mob/user as mob, var/target_zone)
-	if(M == user)
+	if(M == user && !M.check_has_mouth())
```
Comment (NonQueueingMatt) [inline code/game/objects/items/crayons.dm]:
```suggestion
	if(M == user && M.check_has_mouth())
```

**Comment by Doxxmedearly** [comment]
!review

File: code/game/objects/items/crayons.dm (hunk 2 above)
Comment (Doxxmedearly) [reply code/game/objects/items/crayons.dm]: dangit

**Comment by Doxxmedearly** [comment]
!bugfix

File: code/game/objects/items/crayons.dm [hunk 3]
```diff
@@ -105,7 +105,7 @@
 	return
 
 /obj/item/pen/crayon/attack(mob/M as mob, mob/user as mob, var/target_zone)
-	if(M == user)
+	if(M == user && M.check_has_mouth())
 		to_chat(user, "You take a bite of the crayon and swallow it.")
```
Comment (Arrow768) [inline code/game/objects/items/crayons.dm]: Might be worth adding visible messages to that, so the people around can "see" them eating the crayon in the chat.

File: code/game/objects/items/crayons.dm [hunk 4]
```diff
@@ -105,7 +105,7 @@
 	return
 
 /obj/item/pen/crayon/attack(mob/M as mob, mob/user as mob, var/target_zone)
-	if(M == user)
+	if(M == user && M.check_has_mouth())
```
Comment (NonQueueingMatt) [inline code/game/objects/items/crayons.dm]: tthis needs to be casted to human first

**Comment by alsoandanswer** [comment]
> code/game/objects/items/crayons.dm:108:error: M.check_has_mouth: undefined proc

please fix

**Comment by Doxxmedearly** [comment]
Mouth check should be working.
Also added visible messages per Arrow's suggestion

**Comment by alsoandanswer** [comment]
you know, i thought of a funny interaction.

like the IPC just slams the crayon into their screen, shattering the crayon into dust

**Pull request merged**